### PR TITLE
Save coa-eval-override component prior to publishing evaluation

### PR DESCRIPTION
**Pull request by max-hao-d2l**
Related: https://github.com/Brightspace/outcomes-level-of-achievement-ui/pull/58
Unlike the achievements selector in the `consistent-evaluation-outcomes` component, the achievements selector in the `coa-eval-override` component does not autosave, and fires an event with the corresponding "select" action instead.
Changes:
* Add a listener when an outcome in `coa-eval-override` is selected which "unsaves" the evaluation
* Prior to saving/updating/publishing an evaluation, publish the COA demonstration (if applicable)

This should not change any functionality in any tool other than in Mastery View.

### CODE REVIEW

File: components/consistent-evaluation-page.js [hunk 1]
```diff
@@ -220,7 +224,7 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 
 	get _feedbackText() {
 		if (this._feedbackEntity && this._feedbackEntity.properties) {
-			return this._feedbackEntity.properties.html || '';
```
Comment (max-hao-d2l) [inline components/consistent-evaluation-page.js:223]: Unrelated fix: The html field in feedbackEntity in Mastery View is `null`, but we can get text instead.

**Review by rgoessl — APPROVED** [review]

File: components/consistent-evaluation-page.js (hunk 1 above)
Comment (jmurray-d2l) [reply components/consistent-evaluation-page.js:223]: We should probably look into fixing this in the back end.  If there is any styling applied to the feedback in the editor then it won't be returned with `text`

**Comment by jmurray-d2l** [comment]
@AlexBedley Did a bunch of work around the timing of transient saving, looks like you're trying to something similar.  I'd suggest getting his thoughts on this PR before merging

**Comment by AlexBedley** [comment]
> Did a bunch of work around the timing of transient saving, looks like you're trying to something similar. I'd suggest getting his thoughts on this PR before merging

Spoke offline with Max. Gonna look into adding transient save in the API. My fear around doing it this way is that we're moving in the complete opposite direction - the front-end will know less and less as we refactor and just get stuff from the API. So the hack in this PR to make the evaluation unsaved might just be the tip of the iceberg of stuff you'll have to do to make it appear like it's transiently saving.

**Review by gfrancu — APPROVED** [review]

**Comment by max-hao-d2l** [comment]
Given the time constraint, it is probably not feasible to add a new evaluation entity to the evaluation. I've made it so that the COA component autosaves by itself (see https://github.com/Brightspace/outcomes-level-of-achievement-ui/pull/59), and perform a "publish" action on the corresponding entity upon publish/save/update. This prevents us to cache the save action in the frontend.
In the future, we could investigate on making the transient-save and publish action to be similar to Grades and Feedback, as it is most likely cleaner and easier to work with
(Gonna need to fix these failing tests though)

File: components/controllers/ConsistentEvaluationController.js [hunk 2]
```diff
@@ -131,6 +146,7 @@ export class ConsistentEvaluationController {
 			throw new Error(ConsistentEvaluationControllerErrors.INVALID_EVALUATION_ENTITY);
 		}
 
+		if (this.coaDemonstrationHref) await this.saveCoaDemonstration();
```
Comment (AlexBedley) [inline components/controllers/ConsistentEvaluationController.js:149]: nit: i know it's probably the linter's responsibility, but I really think all if-blocks should have brackets `{}`

File: components/consistent-evaluation-page.js (hunk 1 above)
Comment (max-hao-d2l) [reply components/consistent-evaluation-page.js]: I've reverted this change and made the fix in the backend (https://github.com/Brightspace/lms/pull/3185)

Comment (jmurray-d2l) [reply components/consistent-evaluation-page.js]: Nice work Max!

**Review by jmurray-d2l — APPROVED** [review]
LGTM! Would still suggest we wait for Alex to give the green light on this, but up to you.

**Review by AlexBedley — APPROVED** [review]
